jipstatus.py

```diff
@@ -0,0 +1,244 @@
+#!/usr/bin/env python3
+from argparse import ArgumentParser
+from subprocess import call
+from time import gmtime, strftime
+from jinja2 import Template
+
+import json
+import os
+import re
+import sys
+import tempfile
+import yaml
+import datetime
+
+# Local files
+import cfg
+from helper import vprint, eprint
+import jiralogin
+
+import pprint
+pprint = pprint.PrettyPrinter().pprint
+
+def add_domain(user):
+    """
+    Helper function that appends @linaro.org to the username. It does nothing if
+    it is already included.
+    """
+    if '@' not in user:
+        user = user + "@linaro.org"
+    return user
+
+def enumerate_updates(jira):
+    user = cfg.args.user
+
+    since = datetime.datetime.now() - datetime.timedelta(days=int(cfg.args.days))
+
+    jql = "(project = QLT OR assignee in membersOf('linaro-landing-team-qualcomm')) AND updatedDate > -%sd" % cfg.args.days
+    if user:
+       jql += ' AND assignee = "%s"' % add_domain(user)
+    vprint(jql)
+
+    my_issues = jira.search_issues(jql, expand="changelog", fields="summary,comment,assignee,created")
+    if my_issues.total > my_issues.maxResults:
+        my_issues = jira.search_issues(jql, expand="changelog", fields="summary,comment,assignee,created",
+                                       maxResults=my_issues.total)
+
+    for issue in my_issues:
+        changelog = issue.changelog
+        comments = issue.fields.comment.comments
+
+        status = {}
+        status['issue'] = str(issue)
+        if issue.fields.assignee:
+            status['assignee'] = issue.fields.assignee.displayName
+        else:
+            status['assignee'] = 'Unassigned'
+        status['summary'] = issue.fields.summary
+        status['comments'] = []
+        status['resolution'] = None
+
+        created = datetime.datetime.strptime(issue.fields.created, '%Y-%m-%dT%H:%M:%S.%f%z')
+        if created.replace(tzinfo=None) > since:
+            status['resolution'] = 'Created'
+
+        for comment in comments:
+            when = datetime.datetime.strptime(comment.created, '%Y-%m-%dT%H:%M:%S.%f%z')
+            if when.replace(tzinfo=None) < since:
+                continue
+
+            status['comments'].append(comment.body)
+
+        for history in changelog.histories:
+            when = datetime.datetime.strptime(history.created, '%Y-%m-%dT%H:%M:%S.%f%z')
+            if when.replace(tzinfo=None) < since:
+                continue
+            for item in history.items:
+                if item.field == 'resolution':
+                    status['resolution'] = item.toString
+
+        if len(status['comments']) != 0 or status['resolution']:
+            yield(status)
+
+def enumerate_pending(jira):
+    user = cfg.args.user
+
+    since = datetime.datetime.now() - datetime.timedelta(days=7)
+
+    jql = "(project = QLT OR assignee in membersOf('linaro-landing-team-qualcomm')) \
```

Comment:
When I've been testing this patch series I noticed that I'm only able to get output from the QC-team. I think this is the only thing I really would like to see a change to. I.e, either grab the information coming from the user currently logged in and/or add a parameter that takes a (Jira) group as argument.

Reply:
Nice catch Joakim, I'll fix this and add an optional parameter for filtering by project.